Задача №1:   
Когда используется критерий Стьюдента, а когда Z –критерий?

Z-критерий используется когда известна дисперсия   
t-критерий используется когда дисперсия не известна

Задача №2:  
Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм.
Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр
оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.

H0 : 17 = 17.5   
H1 : 17 > 17.5

In [39]:
import scipy.stats as st
import numpy as np

zt = 1.645
a = 0.05
n = 100
zn = (17.5-17)/(4/np.sqrt(n))
print (zn)

1.25


Вывод: 1.25 попадает в область принятия нулевой гипотезы, следовательно гипотеза, что средний диаметр 17 мм при а = 0,05 верна.

Задача №3:  
Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%? (Провести двусторонний тест.)

H0 = 200  
H1 <> 200

In [13]:
x = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
sr = np.mean(x)
sr

198.5

In [14]:
m = np.std(x, ddof=1)
m

4.453463071962462

In [15]:
i = len(x)
i

10

In [22]:
t =(sr-200)/(m/np.sqrt(i))
t

-1.0651074037450896

In [27]:
k = st.t.ppf(0.99, 9)
k

2.8214379233005493

In [35]:
l = 1 - 2*(1 - st.t.cdf(t, 9))
l

-0.6854333964752715

In [36]:
pv = 1 + l
pv

0.31456660352472854

Вывод: Наблюдаемое значение  -1.056 попадает в интервал двухстороннего теста от -2.82 до 2.82 следовательно это нулевая гипотеза (что средний вес пачки печенья составляет 200 г.)

Задача №4:  
Есть ли статистически значимые различия в росте дочерей? 

Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169  
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

Н0 = различий нет
Н1 = различия есть

In [41]:
rm = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169])
rd = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])
res = st.ttest_ind(rm, rd)
res

Ttest_indResult(statistic=0.35581195161541973, pvalue=0.7266292679160495)

In [42]:
kw = st.t.ppf(0.975, 16)
kw

2.1199052992210112

Вывод: поскольку pvalue=0.7266292679160495 больше чем 5% то мы принимает нулевую гипотезу, что статистически значимых различий в росте дочерей нет.  
Наблюдаемое значение 0.355 попадает в интервал двухстороннего теста от -2.11 до 2.11 следовательно это нулевая гипотеза, что статистически значимых различий в росте дочерей нет.